### ISSUE TITLE
specify some layer is untrainable in DAGNN

### ISSUE THREAD

**Issue by pxnguyen** [vlfeat/matconvnet#736]
If I just want to fine-tune only the top three layers in a DAGNN, is there an easy way to achieve this?

**Comment by ankush-me** [comment]
In [+dagnn/Layer.m](https://github.com/vlfeat/matconvnet/blob/master/matlab/%2Bdagnn/Layer.m#L106), the backward pass returns if the gradient of any of its output is empty.

Hence, you could implement a "StopGradient" layer -- with its forward pass as the identity and the backward pass returning an empty array, e.g.:

```
classdef StopGradient < dagnn.ElementWise
  methods
    function outputs = forward(obj, inputs, params)
      outputs = inputs;
    end

    function [derInputs, derParams] = backward(obj, inputs, params, derOutputs)
      derInputs = {} ;
      for i=1:numel(inputs)
        derInputs{end+1} = [];
      end
      derParams = {} ;
    end
  end
end
```

**Comment by pxnguyen** [comment]
Thanks so much. This is a really cool solution.

**Comment by HaziqRazali** [comment]
Wouldn't it be possible to just set the learning rate of all layers except the last 3 to zero ?
